cron.php

```diff
@@ -49,13 +51,177 @@ public function __construct()
+
+        if ( $queueEntities )
+        {
+            foreach ($queueEntities as $entity) 
+            {   switch ($entity->type)
+                {
+                    // delete topic
+                    case FORUM_BOL_UpdateSearchIndexDao::DELETE_TOPIC :
+                        $this->deleteTopicFromSearchIndex($entity->entityId);
+                        break;
+
+                    // update topic
+                    case FORUM_BOL_UpdateSearchIndexDao::UPDATE_TOPIC :
+                        $this->updateTopicInSearchIndex($entity->entityId);
+                        break;
+
+                    // delete group
+                    case FORUM_BOL_UpdateSearchIndexDao::DELETE_GROUP :
+                        $this->deleteGroupFromSearchIndex($entity->entityId);
+                        break;
+
+                    // update group
+                    case FORUM_BOL_UpdateSearchIndexDao::UPDATE_GROUP :
+                        $this->updateGroupInSearchIndex($entity->entityId);
+                        break;
+                }
+
+                $updateSearchIndexDao->delete($entity);
+            }
+        }
+
+        $config->saveConfig('forum', 'update_search_index_cron_busy', 0);
+    }
+
+    /**
+     * Update group
+     * 
+     * @param integer $groupId
+     * @return void
+     */
+    private function updateGroupInSearchIndex( $groupId )
+    {
+        $forumService = FORUM_BOL_ForumService::getInstance();
+        
+        // get the group info
+        $group = $forumService->findGroupById($groupId);
+
+        if ( $group )
+        {
+            // get all topics inrto the group
+            $topics = $forumService->getAllTopicList($groupId);
+
+            if ( $topics )
+            {
+                foreach ($topics as $topic)
+                {
+                    // add the topic
+                    $this->getTextSearchService()->addTopic($topic);
+
+                    // get all posts into the topic
+                    $posts = $forumService->getAllTopicPostList($topic->id);
+
+                    if ( $posts )
+                    {
+                        // add posts
+                        foreach ($posts as $post)
+                        {
+                            $this->getTextSearchService()->saveOrUpdatePost($post, true);
+                        }
+                    } 
+               }
+            }
+        }
+    }
+
+    /**
+     * Delete group from the search index
+     * 
+     * @param integer $groupId
+     * @return void
+     */
+    private function deleteGroupFromSearchIndex( $groupId )
+    {
+        $this->getTextSearchService()->deleteGroup($groupId);
+
+        FORUM_BOL_UpdateSearchIndexDao::getInstance()->
+                addQueue($groupId, FORUM_BOL_UpdateSearchIndexDao::UPDATE_GROUP);
+    }
+
+    /**
+     * Update topic
+     * 
+     * @param integer $topicId
+     * @return void
+     */
+    private function updateTopicInSearchIndex( $topicId )
```

Comment:
Refactor the function to get items with pagination in order to avoid memory limit.